crates/ty/src/lib.rs

```diff
@@ -68,10 +68,12 @@ fn run_check(args: CheckCommand) -> anyhow::Result<ExitStatus> {
 
     let printer = Printer::default().with_verbosity(verbosity);
 
-    tracing::warn!(
-        "ty is pre-release software and not ready for production use. \
+    if std::env::var_os("I_KNOW_TY_IS_PRE_RELEASE").is_none() {
```

Comment:
The environment variable needs. to be added to https://github.com/astral-sh/ruff/blob/35a33f045eecb6652b5c73f82dce26e6c610c3b8/crates/ty_static/src/env_vars.rs#L7